include/circt/Dialect/Verif/VerifOps.td

```diff
@@ -43,6 +47,57 @@ def CoverOp : AssertLikeOp<"cover"> {
   let summary = "Ensure that a property can hold.";
 }
 
+// Edge behavior enum for always block.  See SV Spec 9.4.2.
+
+/// AtPosEdge triggers on a rise from 0 to 1/X/Z, or X/Z to 1.
+def AtPosEdge: I32EnumAttrCase<"Pos", 0, "posedge">;
+/// AtNegEdge triggers on a drop from 1 to 0/X/Z, or X/Z to 0.
+def AtNegEdge: I32EnumAttrCase<"Neg", 1, "negedge">;
+/// AtEdge is syntactic sugar for AtPosEdge or AtNegEdge.
+def AtEdge   : I32EnumAttrCase<"Both", 2, "edge">;
+
+def ClockEdgeAttr : I32EnumAttr<"ClockEdge", "clock edge",
+                                [AtPosEdge, AtNegEdge, AtEdge]> {
+  let cppNamespace = "circt::verif";
+}
+
+class ClockedAssertLikeOp<string mnemonic, list<Trait> traits = []> :
```

Comment:
Do we want to allow clocked assert in procedural regions? I think an assumption that clocked_assert op never be in always statement or if-statement would make the semantics a lot clear? I don't have a strong opinion tough :)

Reply:
Yeah great point. Since these ops define their own clock and enable, they are intended to be used in a non-procedural region.